### PR TITLE
Bug/jenkins 41542 run details stuffs

### PR DESCRIPTION
**Pull request by sophistifunk**
# Description

See [JENKINS-41542](https://issues.jenkins-ci.org/browse/JENKINS-41542).

# Submitter checklist
- [X] Link to JIRA ticket in description, if appropriate.
- [X] Change is code complete and matches issue description
- [ ] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [ ] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.
- [ ] Ran Acceptance Test Harness against PR changes.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

@reviewbybees

### CODE REVIEW

**Comment by sophistifunk** [comment]
screen shot 2017-02-09 at 1 53 46 pm

File: blueocean-dashboard/src/main/js/components/RunDetailsHeader.jsx [hunk 1]
```diff
@@ -77,29 +77,30 @@ class RunDetailsHeader extends Component {
...
                     <ExpandablePath path={ fullDisplayName } hideFirst className="dark-theme" iconSize={ 20 } />
                 </a>
-                <span>#{ run.id }</span>
+                <span>&nbsp;#{ run.id }</span>
             </h1>
         );
 
         const branchSourceDetails = (
-            <div>
+            <div className="u-label-value" title={branchLabel + ': ' + displayName}>
```
Comment (kzantow) [inline blueocean-dashboard/src/main/js/components/RunDetailsHeader.jsx]: NIT: formatting?

File: blueocean-dashboard/src/main/js/components/RunDetailsHeader.jsx [hunk 2]
```diff
@@ -77,29 +77,30 @@ class RunDetailsHeader extends Component {
...
-                         { run.commitId.substring(0, 7) }
-                    </span>
-                </div>
-            );
+        const commitIdString = run.commitId || 'N/A';
+        const commitSourceDetails = (
+            <div className="u-label-value" title={commitLabel + ': ' + commitIdString}>
+                <label>{ commitLabel }:</label>
+                <span className="commit">
+                     { commitIdString.substring(0, 7) }
```
Comment (kzantow) [inline blueocean-dashboard/src/main/js/components/RunDetailsHeader.jsx]: I know this isn't new, but it makes me sad... format should be specified by the scm plugin somehow

File: blueocean-dashboard/src/main/less/run-details-header.less [hunk 3]
```diff
@@ -12,23 +12,41 @@
...
 .RunDetailsHeader-times {
     flex-basis: 20%;
+    max-width: 20%;
+
+    .u-label-value {
+
+        label {
+            display: block;
+            width: 61px;
+            flex-shrink: 0;
```
Comment (kzantow) [inline blueocean-dashboard/src/main/less/run-details-header.less]: ?

File: blueocean-core-js/.eslintrc [hunk 4]
```diff
@@ -5,6 +5,7 @@
     "react/jsx-indent-props": 0,
     "no-unused-vars": [2, {"varsIgnorePattern": "^React$"}],
     "max-len": [1, 160, 4],
+    "prefer-template": 1,
```
Comment (kzantow) [inline blueocean-core-js/.eslintrc]: what does this accomplish?

File: blueocean-dashboard/src/main/js/components/RunDetailsHeader.jsx (hunk 1 above)
Comment (sophistifunk) [inline blueocean-dashboard/src/main/js/components/RunDetailsHeader.jsx]: english

File: blueocean-dashboard/src/main/js/components/RunDetailsHeader.jsx (hunk 2 above)
Comment (sophistifunk) [inline blueocean-dashboard/src/main/js/components/RunDetailsHeader.jsx]: Pull requests welcome ;-)

File: blueocean-core-js/.eslintrc (hunk 4 above)
Comment (sophistifunk) [inline blueocean-core-js/.eslintrc]: Turns off a bullshit lint rule (which is already set to 1 in the root .eslintrc) which conflicts with a different lint rule that's buggy. Using templates inside of JSX code makes the stupid single-quote-only rule shit itself, as well as being harder to read.

File: blueocean-dashboard/src/main/less/run-details-header.less (hunk 3 above)
Comment (sophistifunk) [inline blueocean-dashboard/src/main/less/run-details-header.less]: ...

File: blueocean-core-js/.eslintrc (hunk 4 above)
Comment (kzantow) [inline blueocean-core-js/.eslintrc]: I figured, seeing the rest of it. totally agree: hate template requirement in jsx. 👍

**Comment by i386** [comment]
noice

File: blueocean-dashboard/src/main/less/run-details-header.less (hunk 3 above)
Comment (kzantow) [inline blueocean-dashboard/src/main/less/run-details-header.less]: this is display block. i'm sure there are reasons. i *do* think we should stop using flex for everything, it is a bit buggy in certain situations from what i've seen, but for overall layouts great

Comment (sophistifunk) [inline blueocean-dashboard/src/main/less/run-details-header.less]: In circumstances where it doesn't work, I won't use it. But the labels need to be fixed size if the values are to line up, unless we'd like to further complicate the css with `display:grid`

**Pull request merged**